zetaclient/chains/sui/client/client.go

```diff
@@ -247,6 +250,56 @@ func (c *Client) SuiExecuteTransactionBlock(
 	return parseRPCResponse[models.SuiTransactionBlockResponse]([]byte(resString))
 }
 
+// GetSuiCoinObjectRef returns the latest SUI coin object reference for given owner address
+// Note: the SUI object may change over time, so we need to get the latest object
+func (c *Client) GetSuiCoinObjectRef(ctx context.Context, owner string) (patsui.ObjectRef, error) {
+	coins, err := c.SuiXGetCoins(ctx, models.SuiXGetCoinsRequest{
+		Owner:    owner,
+		CoinType: string(zetasui.SUI),
+	})
+	if err != nil {
+		return patsui.ObjectRef{}, errors.Wrap(err, "unable to get TSS coins")
+	}
+
+	var (
+		suiCoin        *models.CoinData
+		suiCoinVersion uint64
+	)
+
+	// locate the latest version of SUI coin object of given owner
+	for _, coin := range coins.Data {
+		if !zetasui.IsSUICoinType(zetasui.CoinType(coin.CoinType)) {
+			continue
+		}
+
+		version, _ := strconv.ParseUint(coin.Version, 10, 64)
```

Comment:
NIT : Should we continue on error?

Reply:
good catch, should throw error https://github.com/zeta-chain/node/pull/3793/commits/edc687f35edb4abe448647a6adbda8ed4699ddd0

Reply:
Should we continue instead of returning?

Reply:
@ws4charlie, wondering what you think